pypeit/core/scattlight.py

```diff
@@ -0,0 +1,308 @@
+        plt.show()
+        print(res_lsq.x)
+
+        plt.subplot(221)
+        plt.imshow(_frame, vmin=vmin, vmax=vmax)
+        plt.subplot(222)
+        plt.imshow(scatt_img, vmin=vmin, vmax=vmax)
+        plt.subplot(223)
+        plt.imshow(frame - scatt_img, vmin=-vmax/2, vmax=vmax/2)
+        plt.subplot(224)
+        plt.imshow(_frame, vmin=-vmax/2, vmax=vmax/2)
+        # plt.subplot(235)
+        # plt.imshow(scatt_img_alt, vmin=vmin, vmax=vmax)
+        # plt.subplot(236)
+        # plt.imshow(_frame - scatt_img_alt, vmin=vmin, vmax=vmax)
+        plt.show()
+    return scatt_img, res_lsq.x, success
+
+
+def mask_slit_regions(offslitmask, centrace, mask_regions=None):
+    """ Exclude some user-specified inter-slit regions for the fine correction to the scattered light determination
+
+    Parameters
+    ----------
+    offslitmask : `numpy.ndarray`_
+        A boolean mask indicating the pixels that are on/off the slit (True = off the slit)
+    centrace : `numpy.ndarray`_
+        A 2D array, shape is (nspec, nslit), containing the central trace of each slit.
+    mask_regions : :obj:`int`, :obj:`list`, optional
+        A list of regions that should be excluded in the fine correction to the scattered light. A zero
+        value indicates that all pixels left of the first slit will be masked. A value of one indicates
+        that all pixels between the first and second slit will be masked, and so forth. A list of these
+        integers can be supplied. If None, no inter-slit regions will be excluded. If an integer is
+        specified, just one region will be excluded.
+
+    Returns
+    -------
+    good_mask : `numpy.ndarray`_
+        A 2D boolean array indicating the pixels that are suitable to use for the fine scattered light correction.
+    """
+    # Check if there are regions to be masked
+    if mask_regions is None:
+        msgs.warn("There are no inter-slit regions specified that need to be masked")
+        return offslitmask
+    elif isinstance(mask_regions, int):
+        # Convert this to a list
+        _mask_regions = [mask_regions]
+    else:
+        _mask_regions = mask_regions
+
+    # Grab the dimensions
+    nspec, nspat = offslitmask.shape
+    nslit = centrace.shape[1]
+
+    # Setup the pixel coordinates
+    spat = np.arange(nspat)
+
+    # Find the pixels in each slit, limited by the minimum and
+    # maximum spectral position.
+    good_mask = offslitmask.copy()
+    for ii in _mask_regions:
+        if ii == 0:  # All pixels to the left of the first slit
+            indx = spat[None, :] < centrace[:, ii, None]
+        elif ii == nslit:  # All pixels to the right of the last slit
+            indx = spat[None, :] > centrace[:, ii-1, None]
+        else:  # Everything else in between
+            indx = (spat[None, :] > centrace[:, ii-1, None]) \
+                   & (spat[None, :] < centrace[:, ii, None])
+        # Exclude these pixels
```

Comment:
Since offslitmask is boolean, I think the logic would be clearer if you did something like 

good_mask = offslitmask &  np.logical_not(indx)

I would also name indx to indicate what it means, i.e. bad_pixels or something like that.